test/components/component_test.dart

```diff
@@ -14,7 +14,7 @@ void main() {
       expect(c.toPosition().x, 2.2);
       expect(c.toPosition().y, 3.4);
 
-      c.setByPosition(Position(1.0, 0.0));
+      c.setByPosition(Vector2(1.0, 0.0));
```

Comment:
we probably want to rename this to setByVector2 or a better name (maybe setPosition is fine, but setByPosition I was implying that the param is a Position)